test/cpo/CPOClientTest.cpp

```diff
@@ -0,0 +1,147 @@
+/*
+MIT License
+
+Copyright(c) 2022 Futurewei Cloud
+
+    Permission is hereby granted,
+    free of charge, to any person obtaining a copy of this software and associated documentation files(the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and / or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions :
+
+    The above copyright notice and this permission notice shall be included in all copies
+    or
+    substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS",
+    WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+    DAMAGES OR OTHER
+    LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+    SOFTWARE.
+*/
+
+#include <k2/appbase/Appbase.h>
+#include <k2/appbase/AppEssentials.h>
+#include <k2/dto/Timestamp.h>
+#include <k2/tso/client/Client.h>
+#include <k2/cpo/client/Client.h>
+#include <k2/dto/Collection.h>
+#include <k2/dto/ControlPlaneOracle.h>
+
+
+const char* collname = "cpo_client_test_collection";
+
+namespace k2::log {
+inline thread_local k2::logging::Logger cpo_client_test("k2::cpo_client_test");
+}
+
+namespace k2 {
+    using namespace cpo;
+    class CPOClientTest {
+    public:
+        CPOClientTest() {
+            K2LOG_I(k2::log::cpo_client_test, "ctor");
+        }
+        ~CPOClientTest() {
+            K2LOG_I(k2::log::cpo_client_test, "dtor");
+        }
+        seastar::future<> gracefulStop() {
+            K2LOG_I(k2::log::cpo_client_test, "stop");
+            return std::move(_testFuture);
+        }
+        seastar::future<dto::Timestamp> getTimeNow() {
+            return AppBase().getDist<tso::TSOClient>().local().getTimestamp();
+        }
+        seastar::future<> runTest1() {
+            K2LOG_I(k2::log::cpo_client_test, ">>> Test 1: Getting a non-existent collection");
+            return _cpoClient.getPartitionGetterWithRetry(Deadline<>(100ms),String(collname))
+            .then([this] (auto&& response) {
+                auto&[status, resp] = response;
+                K2EXPECT(k2::log::cpo_client_test, status.is4xxNonRetryable(), true);
+            });
+        }
+        seastar::future<> runTest2() {
+            K2LOG_I(k2::log::cpo_client_test, ">>> Test 2: Creating a collection");
```

Comment:
this test is expecting to fail, but it doesn't say why. Could you mention somewhere why this should fail?